central/auth/service/service_impl_test.go

```diff
@@ -126,3 +181,428 @@ func (s *authServiceAccessControlTestSuite) TestAuthServiceResponse() {
 		})
 	}
 }
+
+func (s *authServiceAccessControlTestSuite) TestValidateAuthMachineToMachineConfig() {
+	testCases := []struct {
+		config      *v1.AuthMachineToMachineConfig
+		skipIDCheck bool
+		err         error
+	}{
+		{
+			config: nil,
+			err:    errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{},
+			err:    errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id: "some-id",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "1",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "1s",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "24h1s",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "5m",
+				Type:                    v1.AuthMachineToMachineConfig_GENERIC,
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "5m",
+				Type:                    v1.AuthMachineToMachineConfig_GENERIC,
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "5m",
+				Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole1,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole2,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  "This doesn't exist",
+					},
+				},
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "5m",
+				Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+				Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole1,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole2,
+					},
+					{
+						Key:   "some-key",
+						Value: "some-value",
+						Role:  testRole3,
+					},
+				},
+			},
+		},
+		{
+			config: &v1.AuthMachineToMachineConfig{
+				Id:                      "some-id",
+				TokenExpirationDuration: "5m",
+				Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+				IssuerConfig: &v1.AuthMachineToMachineConfig_GenericIssuerConfig{
+					GenericIssuerConfig: &v1.AuthMachineToMachineConfig_GenericIssuer{
+						Issuer: "https://stackrox.io",
+					},
+				},
```

Comment:
No, it should not. In case of type `GITHUB_ACTIONS`, the generic issuer will simply be ignored (this is also added within the comments of the proto file).

Reply:
Understood, thanks. I overlooked that.